File: SDKs/sportsdata.io/nhl-v3-stats/1.0/go/pkg/models/operations/newsbyplayer.go
package operations




type NewsByPlayerFormatEnum string

const (
    NewsByPlayerFormatEnumXML NewsByPlayerFormatEnum = "XML"
NewsByPlayerFormatEnumJSON NewsByPlayerFormatEnum = "JSON"
)


type NewsByPlayerPathParams struct {
    Format NewsByPlayerFormatEnum `pathParam:"style=simple,explode=false,name=format"`
    Playerid string `pathParam:"style=simple,explode=false,name=playerid"`
    
}

type NewsByPlayerRequest struct {
    PathParams NewsByPlayerPathParams 
    
}

type NewsByPlayerResponse struct {
    ContentType string 
    News []interface{} 
    StatusCode int64 
    
}

File: SDKs/sportsdata.io/nba-v3-stats/1.0/typescript/src/sdk/models/operations/newsbyplayer.ts
import { Metadata, SpeakeasyBase } from "../../../internal/utils/utils";

export enum NewsByPlayerFormatEnum {
    Xml = "XML"
,    Json = "JSON"
}


export class NewsByPlayerPathParams extends SpeakeasyBase {
  @Metadata({ data: "pathParam, style=simple;explode=false;name=format" })
  format: NewsByPlayerFormatEnum;

  @Metadata({ data: "pathParam, style=simple;explode=false;name=playerid" })
  playerid: string;
}


export class NewsByPlayerRequest extends SpeakeasyBase {
  @Metadata()
  pathParams: NewsByPlayerPathParams;
}


export class NewsByPlayerResponse extends SpeakeasyBase {
  @Metadata()
  contentType: string;

  @Metadata()
  news?: any[];

  @Metadata()
  statusCode: number;
}
